primitives/src/opcodes.rs

```diff
@@ -545,10 +545,14 @@ impl Ordinary {
 
 #[cfg(test)]
 mod tests {
-    use std::collections::HashSet;
-
     use super::*;
 
+    #[cfg(feature = "alloc")]
+    use alloc::format;    
+    #[cfg(feature = "alloc")]
+    use alloc::collections::BTreeSet as HashSet;
```

Comment:
Woah! Lets just import `BTreeSet` and change the test to use it. Could be done as a separate patch up front.

Reply:
nit: Technically these imports should be at the top where the `use std::collections::HashSet` line is.